torch/functional.py

```diff
@@ -636,6 +636,41 @@ def cartesian_prod(*tensors):
         return handle_torch_function(cartesian_prod, tensors, *tensors)
     return torch._C._VariableFunctions.cartesian_prod(tensors)
 
+def block_diag(*tensors):
```

Comment:
You might be able to use `torch._C._add_docstr` for this and avoid the python function call overhead. It looks like there aren't any usages in `torch.functional` yet and there might be a good reason for that, so please double check that it works correctly :)

See `torch/nn/functional.py` for usage examples for `_add_docstr`

Reply:
The docstring isn't the only reason why I added this function. I wanted the `tensors` argument to be a variable argument list, rather than an actual list, to match the interface that `scipy.linalg.block_diag` has. I mimicked how `torch.cartesian_prod` implements its variable argument list, but perhaps there's another way to do it which avoids an extra function call?

Reply:
Ah, I missed that, sorry. There may be a way to make the C++ dispatcher emulate argument unpacking like that but if there is someone else will have to point out how.